### DISCUSSION TITLE
MultiSelection usage

### DISCUSSION THREAD

**Discussion by caghetti** [josdejong/svelte-jsoneditor#323] [Q&A]
Hi,

in the documentation the tree mode supports MultiSelection. 
If I understand good the multiselection, it means I can select several elements of my json tree. 
I added `selection: 'MultiSelection',` in my declaration but I have this js error

`index.js:1 Uncaught (in promise) TypeError: Cannot read properties of undefined (reading 'length')`

Did I understood badly the MultiSelection implementation ?

**Comment by josdejong** [comment]
You can indeed select a range of multiple items using `MultiSelection`, you can find the type definition here:

https://github.com/josdejong/svelte-jsoneditor/blob/2699b71f2311d1670e29ccf41848bb283c52121d/src/lib/types.ts#L70-L74

For example:

```js
selection: {
  type: SelectionType .multi,
  anchorPath: ['myArray', 2],
  focusPath: ['myArray', 7],
}
```

To get examples, create your editor with an `onSelect` that logs the actual selection in the developer console, and then manually select multiple items in the editor to see what the selection object becomes.

**Comment by caghetti** [comment]
Thanks for your reply. 
Unfortunately, it's not the the feature I thought about. 
I will describe the feature I thought : I want user could select several items via a ctrl+clic 
Today I saw you can select several lines via shift + clic but my is, as in the file attached, to select only label, startdate and cug

do you think it could be a feature in a next release ?

![image](https://github.com/josdejong/svelte-jsoneditor/assets/148205032/3292e6e8-cea7-4367-80b6-1d8a4b50f32c)

**Reply by josdejong** [reply]
Ah, I understand what you mean.

Currently you can indeed only select a single range with multiple items. It is not possible to select multiple individual items. It would be nice to implement support it though.

**Reply by caghetti** [reply]
Thanks. I will transform this disussion to issue in this case. 
An I forgot to say : your jsoneditor is very incredible : thanks for this editor !

**Reply by josdejong** [reply]
Thanks, glad to hear!
